File: app/src/test/java/me/instabattle/app/managers/TestCallback.java
package me.instabattle.app.managers;

import retrofit2.Call;
import retrofit2.Callback;

import static org.junit.Assert.fail;

abstract class TestCallback<T> implements Callback<T> {
    @Override
    public void onFailure(Call<T> call, Throwable t) {
        fail(t.getMessage());
    }
}

File: app/src/test/kotlin/me/instabattle/app/managers/TestCallback.kt
package me.instabattle.app.managers

import org.junit.Assert
import retrofit2.Call
import retrofit2.Callback

internal abstract class TestCallback<T> : Callback<T> {
    override fun onFailure(call: Call<T>, t: Throwable) {
        Assert.fail(t.message)
    }
}